.github/scripts/instr-count-benchmark/upload.py

```diff
@@ -0,0 +1,121 @@
+"""Scribe Uploader for Pytorch Microbenchmark Data"""
+import argparse
+import json
+import hashlib
+import sys
+import time
+
+import numpy as np
+import torch
+
+from scripts.upload_scribe import ScribeUploader
+
+
+VERSIONS = [
+    'b984f1ebad546ed3ee612244b8c72e53',
+]
+
+
+class PytorchMicrobenchmarkUploader(ScribeUploader):
+    def __init__(self):
+        super().__init__('perfpipe_pytorch_microbenchmarks')
+        self.schema = {
+            'int': [
+                'time',
+                'start_time',
+                'end_time',
+                'benchmark_version',
+                'task_number',
+
+                # We don't need to record as many summary statistics for counts
+                # as we do for times, because we retain them all in
+                # `counts_json`.
+                'count_min', 'count_max',
+                'count_p25', 'count_median', 'count_p75',
+            ],
+            'normal': [
+                # Unique ID for the run.
+                'run_id',
+
+                # Metadata for ad-hoc analysis.
+                'task_key', 'counts_json',
+
+                # TODO: More environment info.
+                'pytorch_version',
+                'pytorch_git_version',
+                'python_version',
+
+            ],
+            'float': [
+                't_min', 't_max', 't_mean',
+                't_p01', 't_p25', 't_median', 't_p75', 't_p99',
+                't_stddev',
+            ]
+        }
+
+    def post_benchmarks(self, result_json):
+        assert result_json['version'] == len(VERSIONS) - 1, \
+            f"Expected version={len(VERSIONS) - 1}, got {result_json['version']}"
+
+        assert result_json['md5'] == VERSIONS[-1], \
+            f"Expected md5={VERSIONS[-1]}, got {result_json['md5']}"
+
+        md5 = hashlib.md5()
+        for key in result_json["values"].keys():
+            md5.update(key.encode("utf-8"))
+        assert result_json['md5'] == md5.hexdigest(), \
+            f"Data reports version MD5 of {result_json['md5']}, however computed {md5.hexdigest()}"
+
+        base_msg = {
+            'time': int(time.time()),
+            'start_time': result_json['start_time'],
+            'end_time': result_json['end_time'],
+            'benchmark_version': result_json['version'],
```

Comment:
Is "benchmark_version" the version of torchbench, or the version of instruction count benchmark? (I guess it is the latter, just wanna confirm). Also, I suggest adding another field called "benchmark_name" and set it to "instruction_count" to make it easier to filter all instruction count data in Scuba.

Reply:
Added a comment and a `benchmark_name` field. (These will live in a separate Scuba table so we don't have to worry about clashing with TorchBench, but still good to be explicit.)